src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -271,109 +276,115 @@ public static BufferedOutputStream buffer(final OutputStream outputStream, final
-     * @throws NullPointerException if the input parameter is null
+     * @param writer the {@code Writer} to wrap or return (not {@code null}).
+     * @return the given {@code Writer} if it is already an instance of {@code BufferedWriter}, or
+     * a new {@code BufferedWriter} for the given {@code Writer}.
+     * @throws NullPointerException if the given {@code Writer} is {@code null}.
      * @since 2.5
      */
     public static BufferedWriter buffer(final Writer writer) {
         return writer instanceof BufferedWriter ? (BufferedWriter) writer : new BufferedWriter(writer);
     }
 
     /**
-     * Returns the given Writer if it is already a {@link BufferedWriter}, otherwise creates a BufferedWriter from the
-     * given Writer.
+     * Returns the given {@link Writer} if it is already a {@link BufferedWriter}, otherwise creates a {@code BufferedWriter}
+     * from the given {@code Writer}, using the specified buffer size.
      *
-     * @param writer the Writer to wrap or return (not null)
-     * @param size the buffer size, if a new BufferedWriter is created.
-     * @return the given Writer or a new {@link BufferedWriter} for the given Writer
-     * @throws NullPointerException if the input parameter is null
+     * @param writer the {@code Writer} to wrap or return (not {@code null}).
+     * @param size the buffer size, if a new {@code BufferedWriter} is created.
+     * @return the given {@code Writer} if it is already an instance of {@code BufferedWriter}, or
+     * a new {@code BufferedWriter} for the given {@code Writer}.
+     * @throws NullPointerException if the given {@code Writer} is {@code null}.
      * @since 2.5
      */
     public static BufferedWriter buffer(final Writer writer, final int size) {
         return writer instanceof BufferedWriter ? (BufferedWriter) writer : new BufferedWriter(writer, size);
     }
 
     /**
-     * Returns a new byte array of size {@link #DEFAULT_BUFFER_SIZE}.
+     * Returns a new {@code byte[]} array of size {@link #DEFAULT_BUFFER_SIZE}.
      *
-     * @return a new byte array of size {@link #DEFAULT_BUFFER_SIZE}.
+     * @return a new {@code byte[]} array of size {@link #DEFAULT_BUFFER_SIZE}.
      * @since 2.9.0
      */
     public static byte[] byteArray() {
         return byteArray(DEFAULT_BUFFER_SIZE);
     }
 
     /**
-     * Returns a new byte array of the given size.
+     * Returns a new {@code byte[]} array of the given size.
      *
      * TODO Consider guarding or warning against large allocations...
      *
      * @param size array size.
-     * @return a new byte array of the given size.
+     * @return a new {@code byte[]} array of the given size.
      * @since 2.9.0
      */
     public static byte[] byteArray(final int size) {
         return new byte[size];
     }
 
     /**
-     * Returns a new char array of size {@link #DEFAULT_BUFFER_SIZE}.
+     * Returns a new {@code char[]} array of size {@link #DEFAULT_BUFFER_SIZE}.
```

Comment:
This is no good IMO: "a a new char array" is clear and correct. "a new {@code char} array" would also be correct. BUT "a "new {@code char[]} array" reads like an array of array. So let's pass on this.